ts/main.ts

```diff
@@ -0,0 +1,34 @@
+const $img = document.querySelector('img');
+const $url = document.querySelector('#user-url') as HTMLInputElement;
+if (!$img) throw new Error('img query failed');
+if (!$url) throw new Error('user-url query failed');
+
+function input(): void {
+  if (!$img) throw new Error('img query failed');
+  $img.src = $url.value;
+}
+
+$url.addEventListener('input', input);
+
+const $form = document.querySelector('#form') as HTMLFormElement;
+if (!$form) throw new Error('form query failed');
+
+interface Input {
+  entryId: number;
+  note: string;
+}
+
+function submit(event: Event): void {
+  event?.preventDefault();
+
+  const formInput: Input = {
+    entryId: data.nextEntryId,
+    note: $form.value,
+  };
```

Comment:
Missing some properties. You should be storing all of the data from the form. So you should have 2 more properties.